pkg/kubernetes/events/event_pubsub.go

```diff
@@ -36,6 +36,29 @@ func (c *osmPubsub) Publish(message PubSubMessage) {
 	c.pSub.Pub(message, message.AnnouncementType.String())
 }
 
+// Unsub is the Unsub implementation for PubSub.
+// It is synchronized, upon exit the channel is guaranteed to be both
+// unsubbed to all topics and closed.
+// This is a necessary step to guarantee garbage collection
```

Comment:
At what point in the lifecycle of the controller will this be used? I see that this is needed for garbage collection, but when will it be used, by who, and why?

Reply:
> At what point in the lifecycle of the controller will this be used?
None currently. It's a good observation, none of the used objects are freed during the lifetime of the controller.
> when will it be used, by who, and why?
This will be immediately used by unit tests. If we don't release the channels on the pubsub backend, they are still hold and subscribed to the relevant topics accross different unit tests contexts.
Most of all, I think it's a relevant API to provide. There might be instances/parts of the code that we might have the need to release a certain pubsub resource, and we must be able to do it properly without leaking or leaving stuff on the pubsub backend.

Reply:
Thanks for the explanation!  Might be good to drop a comment for clarity. LGTM